### PR TITLE
Doc cleanup

### LINKED ISSUE #490: [FEATURE] Documentation cleanup
## Describe the solution you'd like

With the recent cleanup to the examples (#446) and Docker dev environment (#303), the next area that needs cleaning is the documentation (i.e. READMEs).

Right now the documentation does not read well as there have been many authors with different styles, and each README may not be focused.

All of the READMEs should be reviewed for the following:
* Grammatical errors
* Consistent voice (should be cohesive like a single person wrote it)
* Duplication of content (avoid duplication and refer to another README, although may be some cases where we want to keep an entire context together)
* Each file should be focused and contain the basics:
    * README - Overview, including prerequisites, available options/features, and basic usage (quick start guide)
    * CONTRIBUTING - Developer workflows
    * Docker README - How to setup and use Docker dev environment
    * Examples README - How to run the examples
    * Functional Tests README - How to run the functional tests
    * etc

### PR DESCRIPTION
**Pull request by jeremyp-bq**
### Summary

Cleanup markdown doc files.

### Description

- Added a section on release management
- Updated the markdown files to be more consistent in spelling, grammar, code snippets, etc.
- Also fixed a number of minor issues

### Related Issue

Closes #490

### Additional Reviewers

@jonathanl-bq

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -271,19 +265,29 @@ A good pull request means:
...
 
-### Review
 Pull requests are usually reviewed within a few days. If there are comments to address, apply your changes in new commits, rebase your branch and force-push to the same branch, re-run unit and integration suites to ensure tests are still passing. In order to produce a clean commit history, our maintainers would do squash merging once your PR is approved, which means combining all commits of your PR into a single commit in the master branch.
 
 That's it! Thank you for your code contribution!
 
 After your pull request is merged, you can safely delete your branch and pull the changes from the upstream repository.
 
-## Versioning
+### Step 8: Publish a Release
```
Comment (jeremyp-bq) [inline CONTRIBUTING.md:280]: This was the major change for this PR, to add a section on how to publish a new release.

File: README.md [hunk 2]
```diff
@@ -111,93 +116,93 @@ df.write.format("com.vertica.spark.datasource.VerticaSource").options(opts ).mod
...
 
 ### Connector Options
 
 Below is a detailed list of connector options that are used in the options map:
 
-| Configuration Option                           | Possible Values                                     | Description                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       | Required                                                                | Default Value               |
```
Comment (jeremyp-bq) [inline README.md:132]: This is something I wanted to do for a long time.  All the whitespace made it very difficult to maintain this table, so I removed extraneous whitespace and made all the connector options code snippets, making it easier to maintain and read.

File: docs/troubleshooting-guide.md [hunk 3]
```diff
@@ -375,4 +377,4 @@ You may see an error such as:
 21/05/31 17:35:21 ERROR DFSClient: Failed to close inode 16447
 ```
 
-This is simply an issue with hadoop closing file handles on files that have been cleaned up by the connector. It does not cause any functional issues.
\ No newline at end of file
+This is simply an issue with hadoop closing file handles on files that have been cleaned up by the connector. It does not cause any functional issues.
```
Comment (jonathanl-bq) [inline docs/troubleshooting-guide.md:380]: nit: hadoop should be capitalized here

File: functional-tests/README.md [hunk 4]
```diff
@@ -64,44 +68,36 @@ gcs_service_key_id
...
+Note that you should do this outside of the Docker container as it will be extremely slow to compile inside Docker. Navigate to the `functional-tests` folder on your local machine to build the functional test with `sbt assembly`. Since the `spark-connector` folder is mounted onto the containers, the built JAR will also be available on the client container.
 
 To submit the functional test to our standalone cluster, inside the client container navigate to `spark-connector/functional-tests` and use `submit-functional-tests.sh`.
 
-Once submitted, verify through the [web ui](localhost:8080) and the [jobs ui](localhost:4040) that the application was submitted.
-Our functional test, without any arguments, will create multiple spark sessions; You should expect multiple applications executing one after another.
+Once submitted, verify through the [web ui](localhost:8080) and the [jobs ui](localhost:4040) that the application was submitted. Our functional test, without any arguments, will create multiple spark sessions. You should expect multiple applications executing one after another.
```
Comment (jonathanl-bq) [inline functional-tests/README.md:87]: nit: "spark sessions" should be "Spark sessions" probably

**Review by jonathanl-bq — APPROVED** [review]
Left some minor nits, but otherwise looks fine.

**Pull request merged**